apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -65,47 +87,71 @@ def prepare(element_html, data):
             raise Exception("invalid custom format: %s" % custom_format) from None
 
 
-def format_true_ans(element, data, name):
+def format_true_ans(
+    element: lxml.html.HtmlElement, data: pl.QuestionData, name: str
+) -> str:
     a_tru = pl.from_json(data["correct_answers"].get(name, None))
     if a_tru is not None:
         # Get format and comparison parameters
         custom_format = pl.get_string_attrib(element, "custom-format", None)
-        comparison = pl.get_string_attrib(element, "comparison", COMPARISON_DEFAULT)
+        comparison = pl.get_enum_attrib(
+            element, "comparison", ComparisonType, COMPARISON_DEFAULT
+        )
+
         if custom_format is not None:
             a_tru = ("{:" + custom_format + "}").format(a_tru)
-        elif comparison == "relabs":
+        elif comparison is ComparisonType.RELABS:
             # FIXME: render correctly with respect to rtol and atol
             a_tru = "{:.12g}".format(a_tru)
-        elif comparison == "sigfig":
+        elif comparison is ComparisonType.SIGFIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = pl.string_from_number_sigfig(a_tru, digits=digits)
-        elif comparison == "decdig":
+        elif comparison is ComparisonType.DECDIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = "{:.{ndigits}f}".format(a_tru, ndigits=digits)
         else:
-            raise ValueError(
-                'method of comparison "%s" is not valid (must be "relabs", "sigfig", or "decdig")'
-                % comparison
-            )
+            assert_never(comparison)
     return a_tru
 
 
-def render(element_html, data):
+def get_string_precision(number_string: str) -> float:
```

Comment:
(comment so I don't forget) we should have some test cases for these. However, part of this PR might include changing the name of this file so it's easier to write tests. Don't worry quite yet.